state/execution.go

```diff
@@ -119,7 +120,11 @@ func (blockExec *BlockExecutor) CreateProposalBlock(
 
 	preparedProposal, err := blockExec.proxyApp.PrepareProposalSync(
 		context.Background(),
-		abci.RequestPrepareProposal{BlockData: txs.ToSliceOfBytes(), BlockDataSize: maxDataBytes},
+		abci.RequestPrepareProposal{
+      BlockData: txs.ToSliceOfBytes(),
+      BlockDataSize: maxDataBytes,
+      Votes: types.VotesToProto(votes),
+    },
```

Comment:
Unsolved question: How should we be handling the block data size. 

from Marko: Idea, leave it up to application to say how many bytes it wants reaped. Then preprocess proposal will reason about what  it wants to keep if it adds data, etc, or removing excess tx data